File: finstack-py/tests/golden/runners/pricing_cds.py
"""Domain runner for `credit.cds` golden fixtures."""

from __future__ import annotations

from tests.golden.schema import GoldenFixture

from .pricing_common import run_pricing_fixture


def run(fixture: GoldenFixture) -> dict[str, float]:
    """Run one CDS fixture through the Python bindings."""
    return run_pricing_fixture(fixture)

File: finstack/valuations/tests/golden/runners/pricing_cds.rs
//! Domain runner for `credit.cds` golden fixtures.

use crate::golden::runner::DomainRunner;
use crate::golden::schema::GoldenFixture;
use std::collections::BTreeMap;

/// CDS pricing golden runner.
pub struct CdsRunner;

impl DomainRunner for CdsRunner {
    fn run(&self, fixture: &GoldenFixture) -> Result<BTreeMap<String, f64>, String> {
        crate::golden::runners::pricing_common::run_pricing_fixture(fixture)
    }
}
